src/Avalonia.Base/Layout/ContainerType.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Avalonia.Layout
+{
+    /// <summary>
+    /// Defines how a container is queried.
+    /// </summary>
+    public enum ContainerType
+    {
+        /// <summary>
+        /// The container will not be queries for any container size queries.
+        /// </summary>
+        Normal,
+
+        /// <summary>
+        /// The container can be queried for container size queries for width.
+        /// </summary>
+        Width,
+
+        /// <summary>
+        /// The can be queried for container size queries for height.
+        /// </summary>
+        Height,
+
+        /// <summary>
+        /// The can be queried for container size queries for both width and height.
+        /// </summary>
+        WidthAndHeight
```

Comment:
Several questions here:

 1. Why did you deviate from CSS? Just have a `Size` entry that is for everything Height and Width related
 2. Why are Width and Height entries even provided?
    - As far as I can tell there is no performance impact for both height and width (rather than just one)
    - Under what circumstances would a container only have a height or width? It's always just a size we should care about, right?
    - This is unnecessary complexity as far as I can tell

Reply:
Like I responded earlier, CSS has Size and InlineSize as ContainerTypes. We do not have a concept of inline in Avalonia. Just having one size type would prevent containers from growing in size in cases where its preferable, like in ScrollViewers.
In CSS, InlineSize corresponds to Width in normal layouts, and this allows the Height of the container to grow to fit the content, but will fix the Width on the Container to what the element's parent provides. Size type prevents the Container from growing in both dimensions.
To preserve this behavior while giving the user a choice in which dimension to limit, Width, Height and WidthAndHeight are provided as ContainerType. This isn't unfamiliar to Avalonia users, as Window has SizeToContent enum with the same names.